pkg/kubelet/apis/deviceplugin/v1beta1/api.proto

```diff
@@ -73,10 +73,21 @@ message ListAndWatchResponse {
 	repeated Device devices = 1;
 }
 
+message TopologyInfo {
+    	NUMANode node = 1;
```

Comment:
@kad You mentioned [here](https://github.com/kubernetes/kubernetes/pull/80570#discussion_r317557836) that you would like to see the device plugin interface allow one to associate a single device with multiple NUMA nodes. 

Can you elaborate on the use case here. Also, would making this field a `repeated` field be sufficient for what you have in mind?

Reply:
@klueska there are several scenarios: one example is technology in modern processors that is called sub-NUMA clustering. With it enabled, two memory controllers within one physical CPU started to be seen in Linux kernel as two separate NUMA nodes. For devices attached to PCI buses on this physical CPU, depending on device bandwidth,  this difference is almost negligible, so hint for this device would be to span across both of those NUMA nodes.
Another scenario is FPGA devices that are connected via non-PCI links to the CPUs (so they have direct access to the main memory), so alignment to the memory controller might be not so trivial as just single node.

Making it repeatable potentially could be good enough. So, zero instances of NUMANode might be seen "unknown" and one or more instances can be used to transform to `cpuset.mems` in future.

Reply:
You mean FPGAs ? numa_node will have "-1".

Reply:
`local_cpulist`. It has better granularity and allows to align more precisely compared to single integer field `numa_node`.

Reply:
@cbf123 unless `dev_to_node(dev) == -1)`. And that usually a flag to device plugin to search hardware specific properties.

Reply:
@cbf123 yes, but then it means that device plugin needs to use some other internal knowledge of the device or platform than just generic PCI sysfs entries. PCI is not the only one source of truth about real topology. E.g. for PMEM (persistent memory) will expose it via specific entry under something like this: `/sys/devices/LNXSYSTM:00/LNXSYBUS:00/ACPI0012:00/ndbus0/region1` and then that reported topology value should be transformed to node or cpulist, based on the mode in which memory controllers are operating now (e.g. depends on one or more "nodes" per physical CPU).

All in all, we are talking about interface for device plugin, and from kubelet stand point, it doesn't really matter from where device plugin gets the information. What matters is that DP sends a hint that will help to align other resources (CPUs, mems, other devices). One ideal example: device plugin can hint about usage of CPU cores with higher base frequency, in case where one "NUMA node" on a physical die has cores with different frequency/performance settings...

Ideally, we should have ability for device plugins to expose both preferences on both `cpuset.cpus` and `cpuset.mems`, as in practice those two parameters depends on device's knowledge on what is critical resource: CPU-device or memory-device alignment, or both.

But that's longer story, let's get something minimal merged into k8s now, and then we will build up on it. And for things that we are defining now, let's have interfaces flexible enough to be able to extend it in the future. Like above optional/repeatable `NUMANode` (thanks @klueska) is better than single `int` that was in previous iteration.